### ISSUE TITLE
Invalid generated ScalarConfig type for flow types

### ISSUE THREAD

**Issue by emmenko** [dotansimha/graphql-code-generator#1404]
**Describe the bug**
The generated `ScalarConfig` are invalid for flow types.

**To Reproduce**
Given a scalar type (e.g. `DateTime`), a `DateTimeScalarType` is generated of kind `interface`. However, the `GraphQLScalarTypeConfig` flow type from `graphql` [is a `type`](https://github.com/graphql/graphql-js/blob/d0aaa389bcee3e839f96f901ab56bbccd468c31c/src/type/definition.js#L616-L627), which makes it impossible to extend as an interface.

<img width="667" alt="image" src="https://user-images.githubusercontent.com/1110551/53789550-f39c7380-3f24-11e9-9957-c4d5e0b5994c.png">

For TypeScript, the `GraphQLScalarTypeConfig` is an `interface`, which makes it extendible, but not for FlowType.

**Expected behavior**
I wonder what the purpose of this `ScalarConfig` is in the first place. Is it really necessary to have? I tried to find some explanations in the commit history but the only things I found are:

- #910 
- #917 (specifically https://github.com/dotansimha/graphql-code-generator/commit/bab23e56b20ac47f7a72c5c43f9b55524bbe28a2)
- https://github.com/dotansimha/graphql-code-generator/commit/ad07eaff07714b8f692408891a6f3c16145de927 (but the commit didn't have any useful explanation on what the problem was)

I am also unsure on how the `GraphQLScalarTypeConfig` can be "extended". Being a `type`, it's not really possible to do, even with intersections, especially given that the `name` property is literal.

Example: https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgDiATgIY4AWAigDIDKAxqTKcQCq54DCcAdlAEsA5mAC8YAN4AfML1IBbPAC4wAZwzEBvEVIC+AbkycwAEVIY8bAYsbNWPfsLFEylWrZbtODwSIBkkrIKymAA5GYWVoqhYAboAG6sYIkwAK4hEZbWeB72fL7OEqhgQYoqoQD8FaEANKgGQA

**Environment:**
 - OS: _macOS_
 - Codegen: _latest version `0.18.0`_
 - Node: `11`

---

Maybe I'm doing something wrong or I'm missing something. It would be good to understand the reasons for having that and what are we supposed to do about it.

Many thanks! 🙏

**Comment by dotansimha** [comment]
Hi @emmenko ! Thank you for reporting it. 

The right way to do it in flow is to use spread, like that:
```
type DateTimeScalarConfig = { ...GraphQLScalarTypeConfig, name: 'DateTime' };
```

I fixed it in this PR: https://github.com/dotansimha/graphql-code-generator/pull/1520

**Comment by dotansimha** [comment]
Fix available in 1.0.2.

**Comment by emmenko** [comment]
> The right way to do it in flow is to use spread

Ah good to know! Thanks a lot for the fix 🚀
